### ISSUE TITLE
xUnit2007 code fix leads to CS8920 compiler error

### ISSUE THREAD

**Issue by boerdereinar** [xunit/xunit#2798]
xUnit2007 warns to change `Assert.IsAssignableFrom(typeof(IInterface), new Class());` into `Assert.IsAssignableFrom<IInterface>(new Class());`. However this is impossible and results in a CS8920 compiler error because interfaces with virtual static members cannot be used as a type parameter.

# Repro
```cs
public class Repro
{
	[Fact]
	public void Test()
	{
		// xUnit2007
		Assert.IsAssignableFrom(typeof(IInterface), new Class());

		// Code fix => CS8920
		// Assert.IsAssignableFrom<IInterface>(new Class());
	}

	private interface IInterface
	{
		static abstract void Test();
	}

	private class Class : IInterface
	{
		/// <inheritdoc />
		public static void Test() { }
	}
}
```

**Comment by etherfield** [comment]
I did some research on the issue. A year ago there was a [long discussion](https://github.com/dotnet/csharplang/issues/5955) on whether to disallow static abstract members as type parameters for generics or not, and apparently now it's a part of .NET 7.0, C#11 (and C# 10 preview).

Clearly, .NET Framework does not support it. I think we can take advantage of notifying user about xUnit2007 _if_ there is a static abstract member for any target runtime though. If the construction is not supported, there would be a compile error in the first place, so skipping xUnit2007 doesn't matter.

**Comment by etherfield** [comment]
PR: [167](https://github.com/xunit/xunit.analyzers/pull/167)

**Comment by bradwilson** [comment]
This is still not a mainline language feature as the issue is still open: https://github.com/dotnet/csharplang/issues/4436

As such, I'm reluctant to merge the proposed fix, even though it seems simple enough, because we generally don't track previews for concerns that the proposal might be rejected and/or whatever code we add may be wrong in the face of the evolution of the feature.

**Comment by bradwilson** [comment]
Available in `1.6.0-pre.7`
